pkg/deployer/terraform/parse.go

```diff
@@ -354,20 +354,27 @@ func getServiceDependencyOutputs(
 		return nil, err
 	}
 
-	outputs := make(map[string]parser.OutputState, 0)
+	outputs := make(map[string]parser.OutputState)
+
+	var p parser.ResourceRevisionParser
 
 	for _, r := range dependencyRevisions {
-		revisionOutput, err := parser.ParseStateOutputRawMap(r)
+		osm, err := p.GetOutputsMap(r)
 		if err != nil {
 			return nil, err
 		}
 
-		for n, o := range revisionOutput {
+		for n, o := range osm {
 			if _, ok := dependOutputs[n]; !ok {
 				continue
 			}
 
-			outputs[n] = o
+			// FIXME(thxCode): mitigate parser.OutputState to types.OutputValue.
```

Comment:
Mitigate or migrate? Is it hard to include this refactoring in this pr?

Reply:
updated